rama-pac/src/connector/mod.rs

```diff
@@ -0,0 +1,97 @@
+use crate::pac_parser::ProxyDirective;
+use crate::{pac_fetcher::fetch_pac, pac_parser::parse_pac_file};
+use rama_core::{Context, Service, error::BoxError};
+use rama_http::{Uri, service::client::HttpClientExt};
+use rama_http_types::Request;
+
+pub struct PACConnector<S, W> {
+    pub service: S,
+    pub web_client: W,
+    pub pac_uri: Uri,
+}
+
+impl<S, W> PACConnector<S, W>
+where
+    W: HttpClientExt + Send + Sync + 'static,
+{
+    pub fn new(service: S, web_client: W, pac_uri: Uri) -> Self {
+        Self {
+            service,
+            web_client,
+            pac_uri,
+        }
+    }
+}
+
+impl<S, W, Body> Service<Request<Body>> for PACConnector<S, W>
+where
+    S: Service<Request<Body>, Error: Into<BoxError> + Send + Sync + 'static>,
+    W: HttpClientExt + Send + Sync + 'static,
+    Body: Clone + Send + Sync + 'static,
```

Comment:
Body is never clone. Neither should it. Your connector is not dealing with the actual Http request anyway, what you need to implement is a service as a connection service. For any request which has a target uri, but that doesn't have to be a http request perse. This way can also work for any other protocol that has uris and again it's; not to already serve the input but to just establish connection. The inner service is the actual connector stack establishing the connection.